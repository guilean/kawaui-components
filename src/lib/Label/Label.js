import './Label.scss';

import PropTypes from 'prop-types';
import React from 'react';
import clsx from 'clsx';

const Label = ({ text, error }) => {
	return (
		<label
			data-testid="label"
			className={clsx('label', {
				['label--error']: error,
			})}
		>
			{text}
		</label>
	);
};

Label.propTypes = {
	text: PropTypes.string,
	error: PropTypes.bool,
};

export default Label;
