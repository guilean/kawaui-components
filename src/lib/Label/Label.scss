.label {
	font-size: 11px;
	padding: 5px 11px;
	&--error {
		color: red;
	}
}
