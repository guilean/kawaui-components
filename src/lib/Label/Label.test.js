import { cleanup, render } from '@testing-library/react';

import Label from './Label';
import React from 'react';

describe('<Label/>', () => {
	afterEach(() => {
		cleanup();
	});

	it('snapshot', () => {
		const { getByTestId } = render(<Label />);
		const component = getByTestId('label');
		expect(component).toMatchSnapshot();
	});

	it('should render an input node element', () => {
		const { getByTestId } = render(<Label />);
		const component = getByTestId('label');
		expect(component.nodeName).toBe('LABEL');
	});

	it('should render content when text prop', () => {
		const { getByTestId } = render(<Label text="Dummy" />);
		const component = getByTestId('label');
		expect(component.textContent).toBe('Dummy');
	});
});
