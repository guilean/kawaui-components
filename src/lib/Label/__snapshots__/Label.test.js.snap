// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> snapshot 1`] = `
<label
  class="label"
  data-testid="label"
/>
`;
